### ISSUE TITLE
Maven dependencies are not downloaded under JDK9 with sbt 1.0.3

### ISSUE THREAD

**Issue by unoexperto** [sbt/sbt#3726]
(See the guidelines for contributing, linked above)

## steps

sbt update on following project

```
import sbt._
import sbt.Keys._

lazy val rootApi = (project in file(".")).
  settings(
    scalaVersion := "2.11.11",
    resolvers += "Sonatype OSS Snapshots" at "https://oss.sonatype.org/content/repositories/snapshots",
    libraryDependencies ++= {
      val scalaMockV = "3.3.0"
      Seq(
        "org.scalamock" %% "scalamock-scalatest-support" % scalaMockV % Test
      )
    },

    ivyLoggingLevel := UpdateLogging.Full,
    logLevel := Level.Info,
    persistLogLevel := Level.Debug,
    traceLevel := 10
```

## problem

```
expert@expert-ThinkPad ~/work/sideprojects/sbttest $ sbt update
[info] Loading settings from plugins.sbt ...
[info] Loading project definition from /home/expert/work/sideprojects/sbttest/project
[info] Loading settings from build.sbt ...
[info] Set current project to rootApi (in build file:/home/expert/work/sideprojects/sbttest/)
[info] Updating {file:/home/expert/work/sideprojects/sbttest/}rootApi...
[info] :: resolving dependencies :: rootapi#rootapi_2.11;0.1-SNAPSHOT
[info] 	confs: [compile, runtime, test, provided, optional, compile-internal, runtime-internal, test-internal, plugin, pom, scala-tool]
[info] 	found org.scala-lang#scala-library;2.11.11 in sbt-chain
[info] 	[2.11.11] org.scala-lang#scala-library;2.11.11
[info] 	found org.scala-lang#scala-compiler;2.11.11 in sbt-chain
[info] 	found org.scala-lang#scala-reflect;2.11.11 in sbt-chain
[info] 	[2.11.11] org.scala-lang#scala-reflect;2.11.11
[info] 	found org.scala-lang.modules#scala-xml_2.11;1.0.5 in sbt-chain
[info] 	found org.scala-lang.modules#scala-parser-combinators_2.11;1.0.4 in sbt-chain
[info] 	found jline#jline;2.14.3 in sbt-chain
[info] :: resolution report :: resolve 489ms :: artifacts dl 238ms
[info] 	:: modules in use:
[info] 	jline#jline;2.14.3 from sbt-chain in [scala-tool]
[info] 	org.scala-lang#scala-compiler;2.11.11 from sbt-chain in [scala-tool]
[info] 	org.scala-lang#scala-library;2.11.11 from sbt-chain in [compile, runtime, test, compile-internal, runtime-internal, test-internal, scala-tool]
[info] 	org.scala-lang#scala-reflect;2.11.11 from sbt-chain in [scala-tool]
[info] 	org.scala-lang.modules#scala-parser-combinators_2.11;1.0.4 from sbt-chain in [scala-tool]
[info] 	org.scala-lang.modules#scala-xml_2.11;1.0.5 from sbt-chain in [scala-tool]
[info] 	---------------------------------------------------------------------
[info] 	|                  |            modules            ||   artifacts   |
[info] 	|       conf       | number| search|dwnlded|evicted|| number|dwnlded|
[info] 	---------------------------------------------------------------------
[info] 	|      compile     |   1   |   0   |   0   |   0   ||   1   |   0   |
[info] 	|      runtime     |   1   |   0   |   0   |   0   ||   1   |   0   |
[info] 	|       test       |   2   |   0   |   0   |   0   ||   1   |   0   |
[info] 	|     provided     |   0   |   0   |   0   |   0   ||   0   |   0   |
[info] 	|     optional     |   0   |   0   |   0   |   0   ||   0   |   0   |
[info] 	| compile-internal |   1   |   0   |   0   |   0   ||   1   |   0   |
[info] 	| runtime-internal |   1   |   0   |   0   |   0   ||   1   |   0   |
[info] 	|   test-internal  |   2   |   0   |   0   |   0   ||   1   |   0   |
[info] 	|      plugin      |   0   |   0   |   0   |   0   ||   0   |   0   |
[info] 	|        pom       |   0   |   0   |   0   |   0   ||   0   |   0   |
[info] 	|    scala-tool    |   6   |   0   |   0   |   0   ||   6   |   0   |
[info] 	---------------------------------------------------------------------
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	::          UNRESOLVED DEPENDENCIES         ::
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	:: org.scalamock#scalamock-scalatest-support_2.11;3.3.0: Resolution failed several times for dependency: org.scalamock#scalamock-scalatest-support_2.11;3.3.0 {test=[default(compile)]}:: 
[warn] 	1
[warn] 	1
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 
[warn] 	Note: Unresolved dependencies path:
[warn] 		org.scalamock:scalamock-scalatest-support_2.11:3.3.0 (/home/expert/work/sideprojects/sbttest/build.sbt#L8)
[warn] 		  +- rootapi:rootapi_2.11:0.1-SNAPSHOT
[error] sbt.librarymanagement.ResolveException: unresolved dependency: org.scalamock#scalamock-scalatest-support_2.11;3.3.0: Resolution failed several times for dependency: org.scalamock#scalamock-scalatest-support_2.11;3.3.0 {test=[default(compile)]}:: 
[error] 	1
[error] 	1
[error] 
[error] 	at sbt.internal.librarymanagement.IvyActions$.resolveAndRetrieve(IvyActions.scala:331)
[error] 	at sbt.internal.librarymanagement.IvyActions$.$anonfun$updateEither$1(IvyActions.scala:205)
[error] 	at sbt.internal.librarymanagement.IvySbt$Module.$anonfun$withModule$1(Ivy.scala:229)
[error] 	at sbt.internal.librarymanagement.IvySbt.$anonfun$withIvy$1(Ivy.scala:190)
[error] 	at sbt.internal.librarymanagement.IvySbt.sbt$internal$librarymanagement$IvySbt$$action$1(Ivy.scala:70)
[error] 	at sbt.internal.librarymanagement.IvySbt$$anon$3.call(Ivy.scala:77)
[error] 	at xsbt.boot.Locks$GlobalLock.withChannel$1(Locks.scala:93)
[error] 	at xsbt.boot.Locks$GlobalLock.xsbt$boot$Locks$GlobalLock$$withChannelRetries$1(Locks.scala:78)
[error] 	at xsbt.boot.Locks$GlobalLock$$anonfun$withFileLock$1.apply(Locks.scala:97)
[error] 	at xsbt.boot.Using$.withResource(Using.scala:10)
[error] 	at xsbt.boot.Using$.apply(Using.scala:9)
[error] 	at xsbt.boot.Locks$GlobalLock.ignoringDeadlockAvoided(Locks.scala:58)
[error] 	at xsbt.boot.Locks$GlobalLock.withLock(Locks.scala:48)
[error] 	at xsbt.boot.Locks$.apply0(Locks.scala:31)
[error] 	at xsbt.boot.Locks$.apply(Locks.scala:28)
[error] 	at sbt.internal.librarymanagement.IvySbt.withDefaultLogger(Ivy.scala:77)
[error] 	at sbt.internal.librarymanagement.IvySbt.withIvy(Ivy.scala:185)
[error] 	at sbt.internal.librarymanagement.IvySbt.withIvy(Ivy.scala:182)
[error] 	at sbt.internal.librarymanagement.IvySbt$Module.withModule(Ivy.scala:228)
[error] 	at sbt.internal.librarymanagement.IvyActions$.updateEither(IvyActions.scala:190)
[error] 	at sbt.librarymanagement.ivy.IvyDependencyResolution.update(IvyDependencyResolution.scala:20)
[error] 	at sbt.librarymanagement.DependencyResolution.update(DependencyResolution.scala:56)
[error] 	at sbt.internal.LibraryManagement$.resolve$1(LibraryManagement.scala:38)
[error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$12(LibraryManagement.scala:91)
[error] 	at sbt.util.Tracked$.$anonfun$lastOutput$1(Tracked.scala:68)
[error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$19(LibraryManagement.scala:104)
[error] 	at scala.util.control.Exception$Catch.apply(Exception.scala:224)
[error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$11(LibraryManagement.scala:104)
[error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$11$adapted(LibraryManagement.scala:87)
[error] 	at sbt.util.Tracked$.$anonfun$inputChanged$1(Tracked.scala:149)
[error] 	at sbt.internal.LibraryManagement$.cachedUpdate(LibraryManagement.scala:118)
[error] 	at sbt.Classpaths$.$anonfun$updateTask$5(Defaults.scala:2353)
[error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:44)
[error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:42)
[error] 	at sbt.std.Transform$$anon$4.work(System.scala:64)
[error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:257)
[error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:16)
[error] 	at sbt.Execute.work(Execute.scala:266)
[error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:257)
[error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:167)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:32)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:514)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
[error] 	at java.base/java.lang.Thread.run(Thread.java:844)
[error] (*:update) sbt.librarymanagement.ResolveException: unresolved dependency: org.scalamock#scalamock-scalatest-support_2.11;3.3.0: Resolution failed several times for dependency: org.scalamock#scalamock-scalatest-support_2.11;3.3.0 {test=[default(compile)]}:: 
[error] 	1
[error] 	1
[error] Total time: 1 s, completed Nov 9, 2017, 8:06:07 PM
```

## expectation

SBT should download missing dependencies.

## notes

sbt version: 1.0.3 (it works under 1.0.2 + JDK9 or 1.0.3 + JDK8)
java version "9.0.1"
Java(TM) SE Runtime Environment (build 9.0.1+11)
Java HotSpot(TM) 64-Bit Server VM (build 9.0.1+11, mixed mode)

**Comment by xuwei-k** [comment]
https://github.com/sbt/sbt/issues/3684

**Comment by eed3si9n** [comment]
This is now likely fixed in https://github.com/sbt/ivy/pull/27, which will be included in sbt 1.0.4.
